### PR TITLE
Add docstrings for api/adapter.js public functions

### PR DESCRIPTION
**Pull request by mjdietzx**

### CODE REVIEW

File: api/adapter.js [hunk 1]
```diff
@@ -1542,7 +1620,12 @@ class Adapter extends EventEmitter {
         attribute.value = attribute.value.slice(0, offset).concat(value);
     }
 
-    // Callback signature function(err, state) {}
+    /**
+     * Gets and updates this adapter's state.
```
Comment (mrodem) [inline api/adapter.js]: Consider mentioning that the callback will return an instance of `AdapterState`.

File: api/adapter.js [hunk 2]
```diff
@@ -197,11 +214,20 @@ class Adapter extends EventEmitter {
...
+     *
+     * @returns {string} The public key generated from this adapter's key-pair.
+     */
     computePublicKey() {
         this._generateKeyPair();
         return this._security.generatePublicKey(this._keys.sk).pk;
     }
 
+    /**
+     * Sets this adapter's `_keys` property to null. `this._generateKeyPair()` will generate a new key-pair now.
```
Comment (mrodem) [inline api/adapter.js]: There are implementation details here (private variable/function), that the user of the public API should ideally not need to know about.

How about something like: "Deletes any previously generated key-pair. The next time `computeSharedSecret` or `computePublicKey` is invoked, a new key-pair will be generated and used."

File: api/adapter.js [hunk 3]
```diff
@@ -244,7 +270,25 @@ class Adapter extends EventEmitter {
...
+     *
+     * The serial port will be attempted to be opened with the configured serial port settings in `adapterOptions`.
+     * Available adapter open options:
+     * - {number} baudRate: The baud rate this adapter's serial port should be configured with.
+     * - {string} parity: The parity this adapter's serial port should be configured with.
+     * - {string} flowControl: Whether flow control should be configured with this adapter's serial port.
+     * - {number} eventInterval: Interval to use for sending BLE driver events to JavaScript.
+     *                           If 0 events will be sent as soon as they are received from the BLE driver.
+     * - {string} logLevel: The verbosity of logging the developer wants with this adapter.
+     * - {number} retransmissionInterval:
```
Comment (mrodem) [inline api/adapter.js]: This option does not have a description. Consider adding.

File: api/adapter.js [hunk 4]
```diff
@@ -244,7 +270,25 @@ class Adapter extends EventEmitter {
...
+     * - {number} responseTimeout: Response timeout of the data link layer.
+     * - {boolean} enableBLE: Whether the BLE stack should be initialized and enabled.
+     * @param {AdapterOpenOptions} adapterOptions Options to initialize/open this adapter with.
```
Comment (bencefr) [inline api/adapter.js]: `adapterOptions` should only be `options` to be consitent with code.

**Pull request merged**